### PR TITLE
Feat/2656 user links

### LINKED ISSUE #2656: User links for chats/users

### PR DESCRIPTION
**Pull request by crui3er**
Closes #2656

### CODE REVIEW

File: src/dotnet/Chat.Service/ChatsBackend.cs [hunk 1]
```diff
@@ -123,7 +123,6 @@ public virtual async Task<ApiArray<ChatId>> GetPublicChatIdsFor(PlaceId placeId,
         return sChatIds.Select(x => new ChatId(x)).Where(x => !x.IsPlaceRootChat).ToApiArray();
     }
 
-    // Non-compute methods
```
Comment (frolyo) [inline src/dotnet/Chat.Service/ChatsBackend.cs:126]: Now it's a compute method, so let's add a comment

File: src/dotnet/Chat.Service/UserLinksBackendExt.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+namespace ActualChat.Chat;
+
+internal static class UserLinksBackendExt
+{
+    public static async Task UpdateUserLink(
+        ICommander commander,
```
Comment (frolyo) [inline src/dotnet/Chat.Service/UserLinksBackendExt.cs:6]: maybe make it an extension method?

File: src/dotnet/Chat.Service/ChatsBackend.cs [hunk 3]
```diff
@@ -823,6 +852,10 @@ await dbContext.Roles
 
         Chat ApplyDiff(Chat originalChat, ChatDiff? diff) {
             // Update
+            if (diff?.UserLinkId != null)
+                diff = diff with {
+                    UserLinkId = diff.UserLinkId.Value.ToLower()
```
Comment (frolyo) [inline src/dotnet/Chat.Service/ChatsBackend.cs:858]: is it kind of normalization? then how about an ext method for it?

File: src/dotnet/UI.Blazor.App/Components/ChatSettings/EditChatTypeModalPage.razor.cs [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+            static t => new ComputedState<ComputedModel>.Options() {
+                InitialValue = ComputedModel.Loading,
+                Category = ComputedStateComponent.GetStateCategory(t),
+            });
+
+    protected override async Task<ComputedModel> ComputeState(CancellationToken cancellationToken)
+    {
+        var chatId = ChatId;
+
+        var chat = await Chats.Get(Session, chatId, cancellationToken);
```
Comment (frolyo) [inline src/dotnet/UI.Blazor.App/Components/ChatSettings/EditChatTypeModalPage.razor.cs]: no configure await? intentional?

File: src/dotnet/UI.Blazor.App/Pages/EmbeddedChatPage.razor [hunk 5]
```diff
@@ -102,22 +99,16 @@
 
                 _ = InvokeAsync(async () => {
                     // Set new ChatSid
-                    ChatSid = newChat.Id;
-                    await SelectChat(chatSid);
+                    await SelectChat(newChat.Id);
```
Comment (frolyo) [inline src/dotnet/UI.Blazor.App/Pages/EmbeddedChatPage.razor:102]: no need for async/await?

**Review by frolyo — APPROVED** [review]

**Review by alexis-kochetov — APPROVED** [review]

File: src/dotnet/Chat.Service/ChatsBackend.cs (hunk 1 above)
Comment (crui3er) [reply src/dotnet/Chat.Service/ChatsBackend.cs:126]: Done

File: src/dotnet/Chat.Service/UserLinksBackendExt.cs (hunk 2 above)
Comment (crui3er) [reply src/dotnet/Chat.Service/UserLinksBackendExt.cs:6]: I don't want make it extension method to keep ICommander interface api cleaner.

File: src/dotnet/UI.Blazor.App/Components/ChatSettings/EditChatTypeModalPage.razor.cs (hunk 4 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor.App/Components/ChatSettings/EditChatTypeModalPage.razor.cs]: Fixed.

File: src/dotnet/UI.Blazor.App/Pages/EmbeddedChatPage.razor (hunk 5 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor.App/Pages/EmbeddedChatPage.razor:102]: I didn't get what you mean. Can you restate, please?

File: src/dotnet/Chat.Service/ChatsBackend.cs (hunk 3 above)
Comment (crui3er) [reply src/dotnet/Chat.Service/ChatsBackend.cs:858]: ToLower method is already a normalization method but with more specific name.
What do you suggest to do?

**Pull request closed without merging**